### ISSUE TITLE
Separate Zones based on orientation

### ISSUE THREAD

**Issue by mostaphaRoudsari** [tt-acm/EnergyAnalysisForDynamo#66]
Similar to https://github.com/tt-acm/EnergyAnalysisForDynamo/issues/65 but this time for orientations.

**Comment by mostaphaRoudsari** [comment]
Now that we have decompose zone maybe this should be separate surfaces based on orientation? Then having zones with multiple walls facing different directions will be already addressed!
I will check rest of issues including this one on Monday.

**Comment by mostaphaRoudsari** [comment]
With combination of this node and separate by levels users can do interesting facade studies. Here is an example to change glazing constructions. @arkdanielnielsen I guess you will like this for your studies.

![image](https://cloud.githubusercontent.com/assets/2915573/5826588/d72a8742-a0c1-11e4-97db-df2705873c8a.png)


PS1: I found 2 bugs in Vasari zoning playing around with these new set of nodes:
1. Sometimes it generates window between interior walls.
2. Some of the floors are wrongly assigned as roof.

PS2: DesignScript's vector.AngleBetween doesn't work as expected so I had to write a private method to handle that for now. I've already reported this.
